tests/utils/test_io_utils.py

```diff
@@ -91,3 +92,34 @@ def test_read_cameras_txt_nonexistent_file() -> None:
 
 
 # TODO in future PR: add round-trip test on write poses to images.txt, and load poses (poses->extrinsics->poses)
+
+
+def test_round_trip_images_txt() -> None:
+    """Starts with a pose that was produced after applying an alignment transformation to a camera frustum. Converts
+    sample pose into extrinsics. Then converts these extrinsics back into a pose and verifies it matches
+    with the original."""
+
+    # Setup dummy pinhole camera object
+    default_intrinsics = Cal3Bundler(fx=100, k1=0, k2=0, u0=0, v0=0)
+
+    original_pose = np.array(
+        [
+            [0.0018451, -0.994551, 0.104234, -0.0121756],
```

Comment:
@adisingh50 I checked the determinant of the rotation portion of this matrix and it seems to be -1.

Where did you get this rotation matrix from? It looks like we are dealing with a reflection, not a rotation.

Likely this is the issue

Reply:
Cool. @adisingh50 can you try the following?  Can you check the determinant of V^T that you take out of the SVD, and if it's -1, check which singular value was -1? and then correct that singular value to be +1, and apply it to the corresponding row or column of V? And see if that fixes all our issues?

You can read a bit more about it here under Orthogonal Procrustes:
https://en.wikipedia.org/wiki/Orthogonal_Procrustes_problem#Generalized.2Fconstrained_Procrustes_problems

https://dsp.stackexchange.com/questions/3431/reflection-rotation-matrix

https://towardsdatascience.com/essential-math-for-data-science-visual-introduction-to-singular-value-decomposition-svd-ca549ab7eb9c

Reply:
Yup. I'll get back to you later today about this.